### ISSUE TITLE
Questions on verification_RFW and training procedure 

### ISSUE THREAD

**Issue by lizhenstat** [visionjo/facerec-bias-bfw#61]
Hi, 
Thanks for your great work and sharing of the code on these two papers !
It takes me days to read the paper and go through the repository and I have a few questions:

(1) from readme.txt in folder Accuracy_for_RFW, we should put "verification_RFW.py in the following folder: ./src/eval/" in insightFace repo but I cannot find this repo ( I assume it is because insightFace may have changed over the past year, can you tell me which version of insightFace is compatible with verification_RFW.py since I want to test that, thanks) 

(2) Do you have the code for training the features (asian_females, asian_males, black_females, black_males, indian_females, indian_males,...). Since I have a hard time finding something like train.py (e.g. the loss function and training process).
(I suppose the released code is mainly on image pre-processing and result analysis)
(Since BFW dataset is not as large as other face dataset and it may possible for me to train it from scratch on one GPU)

Thanks for your time and any help would be appreciated !

**Comment by visionjo** [comment]
Thank you for your reply; also, for expressing your interest. All much appreciated.

I have been a little busy this week, but by the EOD Friday, I will get back to you on this! Just wanted to give you a heads up in meantime.

Cheers,
Joe


--

Joseph Robinson, Ph.D.
Email: ***@***.******@***.***>
Website: www.jrobsvision.com<http://www.jrobsvision.com/>

Follow me on ***@***.***>!

Cell: +1 (978) 764-0003

________________________________
From: lizhen ***@***.***>
Sent: Tuesday, August 10, 2021 4:05 AM
To: visionjo/facerec-bias-bfw ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [visionjo/facerec-bias-bfw] Questions on verification_RFW and training procedure (#61)


Hi,
Thanks for your great work and sharing of the code on these two papers !
It takes me days to read the paper and go through the repository and I have a few questions:

(1) from readme.txt in folder Accuracy_for_RFW, we should put "verification_RFW.py in the following folder: ./src/eval/" in insightFace repo but I cannot find this repo ( I assume it is because insightFace may have changed over the past year, can you tell me which version of insightFace is compatible with verification_RFW.py since I want to test that, thanks)

(2) Do you have the code for training the features (asian_females, asian_males, black_females, black_males, indian_females, indian_males,...). Since I have a hard time finding something like train.py (e.g. the loss function and training process).
(I suppose the released code is mainly on image pre-processing and result analysis)
(Since BFW dataset is not as large as other face dataset and it may possible for me to train it from scratch on one GPU)

Thanks for your time and any help would be appreciated !

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvisionjo%2Ffacerec-bias-bfw%2Fissues%2F61&data=04%7C01%7Crobinson.jo%40northeastern.edu%7C35d3fad0dca843720b6308d95bd5a3db%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C637641795404517748%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yQ6cOOLmwRRdpr725kSdZ0%2FqjYTx3oGnDxzVbbOIa%2FM%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABAGNZMPTBJ5WTLQO6XNC2TT4DMVDANCNFSM5B3TPMSA&data=04%7C01%7Crobinson.jo%40northeastern.edu%7C35d3fad0dca843720b6308d95bd5a3db%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C637641795404527744%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=L2vwjqyYqUZXhxWjjRneJeDMS0EYDh6%2BBmAbN93imvw%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Crobinson.jo%40northeastern.edu%7C35d3fad0dca843720b6308d95bd5a3db%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C637641795404537735%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yxr6YxgPdegxznt5Nx52KX0Mk60%2BFhfesn1wqIWbAgk%3D&reserved=0> or Android<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7Crobinson.jo%40northeastern.edu%7C35d3fad0dca843720b6308d95bd5a3db%7Ca8eec281aaa34daeac9b9a398b9215e7%7C0%7C0%7C637641795404537735%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FLaFZ1%2BR%2BTMVKnPhTwbYZdfNt4rkTI7I%2FnDWdsPXe%2FE%3D&reserved=0>.

**Comment by lizhenstat** [comment]
@visionjo hi, thanks for your reply!
Besides, I also found the pictures in the following folder:
"jrobby\bfw\bfw-cropped-aligned\asian_females\n009042"
the following three pictures(0015_01.jpg, 0034_01.jpg, 0394_01.jpg) belong to **Guo Jing Jing** while folder "n009042" is of **Wu Min Xia,** both of them are famous Diving Athlete in China, 
![0034_01](https://user-images.githubusercontent.com/41462408/129690875-0e13a081-2ca4-4d7b-8367-a9c5bf0f9e6e.jpg)![0394_01](https://user-images.githubusercontent.com/41462408/129690919-98f1c8d9-bf14-417f-b447-36799b703689.jpg)![0394_01](https://user-images.githubusercontent.com/41462408/129690983-9c3ef773-3e7a-412c-a631-6ad6dd19f223.jpg)

I think there are many picutures of them together from the internet then there maybe some confuse.
![guojingjing](https://user-images.githubusercontent.com/41462408/129692334-55954f0f-9ed6-4d53-9676-64551a5eb008.jpg)

Here is the snapshot of the folder "n009042":
![snapshot](https://user-images.githubusercontent.com/41462408/129692088-e5105903-fad6-40fe-bbd3-7c6fbac22fc9.jpg)

**Comment by visionjo** [comment]
Thanks for pointing this out. We are currently integrating the data as part of the fiftyone datasets, and as part of this an updated version of data will be available. This bug will be addressed and released with this ASAP (within weeks). Thanks again and do post additional items that come up 👌

**Comment by visionjo** [comment]
I'll provide the features and answer your original question sooner (a night this week). My apologies, I lost track of this. Thanks for reaching back out.

**Comment by lizhenstat** [comment]
@visionjo Hi, thanks, can you notice me if you update the dataset. Thanks a lot !
